src/glossary.ts

```diff
@@ -192,9 +192,9 @@ export type Value<
     ? ReturnType<Target[Key]['getValue']>
     : // Extract value type from relations.
     Target[Key] extends OneOf<infer ModelName>
-    ? PublicEntity<Dictionary, ModelName>
+    ? PublicEntity<Dictionary, ModelName> | undefined
     : Target[Key] extends ManyOf<infer ModelName>
-    ? PublicEntity<Dictionary, ModelName>[]
+    ? PublicEntity<Dictionary, ModelName>[] | undefined
```

Comment:
Do we need to explicitly add `undefined` to this type? 

The initial values for the entities are always `Partial<Value<...>>`, which implies that its properties are only a subset of the model properties, making each property potentially undefined. Was there some use-case that made you add this change, @olivierwilkinson?

Reply:
Is this change aimed exclusively at the proper annotation of the `post.author` value when inspecting it (as in, TypeScript says it's an existing relationship when it can also be undefined)?

Reply:
Yes exactly it's for proper annotation of the `post.author` value. It cropped up when using entities as response data in msw graphql resolvers. For example:

```typescript
db.post.create() // post.author is undefined

type GetPost = {
  post: {
    id: string
    author: {
      id: string
    }
  }
}

server.use(
  graphql.query<GetPost, { id: string }>('GetPost', (req, res, ctx) => {
    const post = db.post.findFirst({
      where: { id: { equals: req.variables.id } },
      strict: true,
    })!

    // Typescript doesn't warn us that `post` doesn't match the GetPost type
    return res(ctx.data({ post })); 
  })
)
```

Reply:
Got it! Thanks.